server/planning/planning_locks.py

```diff
@@ -1,45 +1,38 @@
-# -*- coding: utf-8; -*-
-#
-# This file is part of Superdesk.
-#
-#  Copyright 2023 Sourcefabric z.u. and contributors.
-#
-# For the full copyright and license information, please see the
-# AUTHORS and LICENSE files distributed with this source code, or
-# at https://www.sourcefabric.org/superdesk/license
-
-from enum import Enum
+import asyncio
+from enum import Enum, unique
+from typing import Annotated, Any
 
 from eve.utils import ParsedRequest
-from eve.render import send_response
+from pydantic import BaseModel, Field, field_validator
 
 from superdesk.core import json
-from superdesk.flask import request, Blueprint
-from superdesk import Resource, get_resource_service, blueprint
-from superdesk.auth.decorator import blueprint_auth
-
+from superdesk import get_resource_service
+from superdesk.core.web import EndpointGroup
+from superdesk.core.types import Response, SearchRequest, SearchArgs, ESQuery
+
+from planning.events import EventsAsyncService
+from planning.planning import PlanningAsyncService
+from planning.assignments import AssignmentsAsyncService
+from planning.core.service import BasePlanningAsyncService
 from planning.utils import get_first_related_event_id_for_planning
 from planning.search.queries.elastic import ElasticQuery, field_exists
 
 
-class PlanningLocksResource(Resource):
-    resource_methods = ["GET"]
-    item_methods = []
-    endpoint_name = "planning_locks"
-    allow_unknown = True
+planning_locks_endpoints = EndpointGroup("/planning_locks", __name__, url_prefix="api")
```

Comment:
The `url_prefix` should not be here, as it's controlled via the app factory's [register_endpoint](https://github.com/superdesk/superdesk-core/blob/async/superdesk/factory/app.py#L302) function, and the [api_prefix](https://github.com/superdesk/eve/blob/sync-upstream/eve/flaskapp.py#L610) property

Reply:
Do we really need an EndpointGroup here anyway, considering we're registering a single endpoint?

Reply:
> as it's controlled via the app factory's [register_endpoint](https://github.com/superdesk/superdesk-core/blob/async/superdesk/factory/app.py#L302) function, and the [api_prefix](https://github.com/superdesk/eve/blob/sync-upstream/eve/flaskapp.py#L610) property

right, that's what I thought but, for some reason I does not pick it up. Do you think I might have some missing configuration/setting? I've checked the URL_PREFIX env var and it's there so I'm not sure what could be.

Reply:
> Do we really need an EndpointGroup here anyway, considering we're registering a single endpoint?

indeed, we do not need a group in this case. I'll change it.